### PR TITLE
Panda, Tiger and Eagle

### PR DESCRIPTION
**Pull request by hanloong**
Here my assignment for the 3 components.

thanks in advance

### CODE REVIEW

File: lib/movie_library.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  def slope
+    sorted_movies = sort_movies
+    return 0 if sorted_movies.last.year == sorted_movies.first.year
+    (sorted_movies.last.score - sorted_movies.first.score).to_f / (sorted_movies.last.year - sorted_movies.first.year).to_f
+  end
+
+  private
+  def sort_movies
+    @movies.sort_by { |m| m.year }
```
Comment (jwo) [inline lib/movie_library.rb:30]: really like the use of private here :thumbsup:

File: spec/movie_library_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  let(:donnie) { double("movie", title: "Donnie Darko", year: 2002, score: 100) }
+
+  it "should store movies" do
+    library.add(empire)
+
+    expect(library.movies.first).to eq(empire)
+  end
+
+  it "should be able to store multiple movies" do
+    library.add(empire)
```
Comment (jwo) [inline spec/movie_library_spec.rb:15]: fantastic way to test here with real objects

File: spec/api_spec.rb [hunk 3]
```diff
@@ -24,4 +24,16 @@
...
+
+  it "should not raise error if not found" do
+    expect {
+        Api.search_by_title("NOTHINGFOUNDHERE")
+
+    }.to_not raise_error
+  end
+
+  it "should return false if searching for blank" do
+    expect(Api.search_by_title("")).to be_false
```
Comment (jwo) [inline spec/api_spec.rb:36]: appropriate use of the expect syntax (rather than the .should()).

**Comment by jwo** [comment]
Fantastic job here. I especially liked:
1. the MovieLibrary class
2. the specs you added

Well done, can't wait to see more!

**Comment by hanloong** [comment]
thanks for the feedback @jwo

**Pull request closed without merging**